bin/quipucords-installer

```diff
@@ -408,7 +444,61 @@ remove_container_images() {
   fi
 }
 
+upgrade_with_db() {
+  echo "Upgrading Quipucords and database ..."
+  local upgrade_ts db_backup_filename db_backup_file
+  upgrade_ts="$(date '+%FT%T')"
+  db_backup_filename="quipucords_db_backup.${upgrade_ts}.sql"
+  db_backup_file="${QUIPUCORDS_DATA_DIR}/db/${db_backup_filename}"
+
+  echo "Backing up Quipucords database to ${db_backup_file} ..."
+  # Database needs to be running for getting a backup
+  systemctl --user start quipucords-db
+  podman exec -t quipucords-db /usr/bin/pg_dumpall >"${db_backup_file}"
+
+  stop_containers
+
+  local userdata_dir="${QUIPUCORDS_DATA_DIR}/db/userdata"
+  # Save the current database
+  echo "Saving the Quipucords database to ${userdata_dir}.${upgrade_ts} ..."
+  mv "${userdata_dir}" "${userdata_dir}.${upgrade_ts}"
+  mkdir "${userdata_dir}"
+
+  install
+  pull_latest_images
+
+  # Start the Quipucords DB for restoring from the backup
+  echo "Starting Quipucords DB ..."
+  systemctl --user start quipucords-db
+  wait_for_db
+
+  echo "Restoring Quipucords database from ${db_backup_file} ..."
+  local db_env db_user db_name
+  db_env="${QUIPUCORDS_ENV_DIR}"/env-db.env
+  db_user="$(grep -o '^QUIPUCORDS_DBMS_USER=.*' "${db_env}" | tail -n 1 | sed 's/^[^=]\+=//')"
+  db_name="$(grep -o '^QUIPUCORDS_DBMS_DATABASE=.*' "${db_env}" | tail -n 1 | sed 's/^[^=]\+=//')"
+  podman exec -it quipucords-db /usr/bin/psql -U "${db_user}" -d "${db_name}" \
+    -f "/var/lib/pgsql/data/${db_backup_filename}"
+  return $?
+}
+
 upgrade() {
+  local db_container="quipucords-db.container"
+  local from_db_image
+  from_db_image=$(get_image "${SYSTEMD_CONFIG_DIR}/${db_container}")
```

Comment:
Not sure if we can do it in this context, but I think we should generally prefer `systemctl show`, which displays settings that systemd actually uses - after merging all configuration sources and resolving all drop-ins.